### ISSUE TITLE
BBANDS width are same across all matypes

### ISSUE THREAD

**Issue by paritoshjchandran** [mrjbq7/ta-lib#165]
```python
import numpy as np
import talib
from talib import MA_Type


a=[37425.0, 42750.0, 34500.0, 39750.0, 46050.0, 50625.0, 54375.0, 50475.0, 53775.0, 57450.0, 56175.0, 65775.0, 66975.0, 67575.0, 66225.0, 72525.0, 71925.0, 70800.0, 70050.0, 71850.0, 70050.0, 67125.0, 67650.0, 73800.0, 73500.0, 65850.0, 59475.0, 51225.0, 63300.0, 63300.0, 59550.0, 65550.0, 59550.0, 60525.0, 67800.0, 72225.0, 63300.0, 65325.0, 69375.0, 71025.0, 71700.0, 66300.0, 68100.0, 65550.0, 67650.0, 69375.0, 68775.0, 65400.0, 65100.0, 66975.0, 68400.0, 66150.0, 64575.0, 66000.0, 68175.0, 64500.0, 68400.0, 69450.0, 68100.0, 66675.0, 63975.0, 66000.0, 63150.0, 63075.0, 68850.0, 74100.0, 66825.0, 72450.0, 69300.0, 72600.0, 70950.0, 72300.0, 72300.0, 73275.0, 67800.0, 66525.0, 70425.0, 67950.0, 65175.0, 64725.0, 66000.0, 68550.0, 72075.0, 73500.0, 73800.0, 74700.0, 73725.0, 75975.0, 71700.0, 62850.0, 61200.0, 65550.0, 66450.0, 67725.0, 67500.0, 72525.0, 73650.0, 77475.0, 79050.0, 78600.0, 84300.0, 84750.0, 85350.0, 86250.0, 83850.0, 85050.0, 84300.0, 86250.0, 87000.0, 87375.0, 90900.0, 89475.0, 91275.0, 84825.0, 81225.0, 82800.0, 83850.0, 82725.0, 78000.0, 73575.0, 76050.0, 79500.0, 76350.0, 72600.0, 74625.0, 75450.0, 78375.0, 78450.0, 85650.0, 84975.0, 85500.0, 85950.0, 89250.0, 84825.0, 83175.0, 84225.0, 84225.0, 85050.0, 87075.0, 87075.0, 85875.0, 87075.0, 87150.0, 85350.0, 87375.0, 87225.0, 86250.0, 81525.0, 80625.0, 81525.0, 82350.0, 80100.0, 78300.0, 76500.0, 74700.0, 75450.0, 75150.0, 74850.0, 79500.0, 76350.0, 77625.0, 76575.0, 75525.0, 75075.0, 74400.0, 76725.0, 75000.0, 74925.0, 75600.0, 78000.0, 79350.0, 77550.0, 78225.0, 76725.0, 78150.0, 74775.0, 77775.0, 78675.0, 81450.0, 81225.0, 79725.0, 79725.0, 83025.0, 80100.0, 82200.0, 85050.0, 84150.0, 86325.0, 87375.0, 85500.0, 85650.0, 84300.0, 83250.0, 80100.0, 83400.0, 81000.0, 82350.0, 80550.0, 81900.0, 81525.0, 83775.0, 85650.0, 87900.0, 85575.0, 82425.0, 81825.0, 81975.0, 82575.0, 82125.0, 76950.0, 80850.0, 81000.0, 72825.0, 80250.0, 79650.0, 78600.0, 80175.0, 79500.0, 79500.0, 82200.0, 82950.0, 84225.0, 79500.0, 83025.0, 82500.0, 81900.0, 74100.0, 75150.0, 74550.0, 71550.0, 77925.0, 81225.0, 75975.0, 77850.0, 74550.0, 71550.0, 76875.0, 73650.0, 73575.0, 72750.0, 74925.0, 77700.0, 75375.0, 75150.0, 76575.0, 76725.0, 76200.0, 73950.0, 76350.0, 71775.0, 73650.0, 72900.0, 70500.0, 68100.0, 70575.0, 72150.0, 71550.0, 70950.0, 72600.0, 69975.0, 68250.0, 69600.0, 71250.0, 69000.0, 70950.0, 67575.0, 69000.0, 73650.0, 72450.0, 72675.0, 71325.0, 67875.0, 66750.0, 71700.0, 69375.0, 71775.0, 70500.0, 66450.0, 70500.0, 67425.0, 63750.0, 62325.0, 61725.0, 64425.0, 62475.0, 61800.0, 68250.0, 66675.0, 64425.0, 63900.0, 66675.0, 67125.0, 66450.0, 64425.0, 69750.0, 65850.0, 70950.0, 71250.0, 71775.0, 73500.0, 74250.0, 74925.0, 72375.0, 77325.0, 75225.0, 74625.0, 77100.0, 73575.0, 77700.0, 80700.0, 78300.0, 75375.0, 74175.0, 74475.0, 75675.0, 76275.0, 78150.0, 78375.0, 79500.0, 83100.0, 82350.0, 80550.0, 79650.0, 80850.0, 76575.0, 78750.0, 76950.0, 77400.0, 79500.0, 79500.0, 76425.0, 78675.0, 77850.0, 79050.0, 78075.0, 77025.0, 79875.0, 78450.0, 79425.0, 78300.0, 81075.0, 82050.0, 82350.0, 80700.0, 80325.0, 79350.0, 78375.0, 78300.0, 78675.0, 74550.0, 74700.0, 76725.0, 75900.0, 76875.0, 76500.0, 76875.0, 73200.0, 74025.0, 75825.0, 81525.0, 82575.0, 82950.0, 87075.0, 85725.0, 83175.0, 84750.0, 85425.0, 86550.0, 88650.0, 84600.0, 84525.0, 81150.0, 81975.0, 88050.0, 85950.0, 73350.0, 54825.0, 72900.0, 71250.0, 76050.0, 69900.0, 71100.0, 68550.0, 67050.0, 70425.0, 71400.0, 72150.0, 72000.0, 74325.0, 72450.0, 69750.0, 71700.0, 62925.0, 69525.0, 67500.0, 67575.0, 68550.0, 57000.0, 55950.0, 52725.0, 54075.0, 55050.0, 54300.0, 49275.0, 47775.0, 50475.0, 51600.0, 54750.0, 55500.0, 55725.0, 53175.0, 52950.0, 54900.0, 56775.0, 57600.0, 63000.0, 61725.0, 57525.0, 59925.0, 59175.0, 63900.0, 61275.0, 60300.0, 54375.0, 50850.0, 57075.0, 53325.0, 55875.0, 55275.0, 52500.0, 61200.0, 61275.0, 69825.0, 72975.0, 66975.0, 66375.0, 65700.0, 66300.0, 68550.0, 66750.0, 65700.0, 63825.0, 62175.0, 60375.0, 60675.0, 59550.0, 58500.0, 62100.0, 64875.0, 57075.0, 61650.0, 64575.0, 66900.0, 67050.0, 65550.0, 65625.0, 65550.0, 63450.0, 61725.0, 64425.0, 58725.0, 60300.0, 60750.0, 59775.0, 61200.0, 59475.0, 59100.0, 59250.0, 59025.0, 59550.0, 59925.0, 60675.0, 64125.0, 62325.0, 61500.0, 59475.0, 60900.0, 58950.0, 60300.0, 62925.0, 63375.0, 63750.0, 67200.0, 67950.0, 68550.0, 66300.0, 66000.0, 67050.0, 63900.0, 64350.0, 63750.0, 64950.0, 64800.0, 73425.0, 72825.0, 70500.0, 71400.0, 71175.0, 69300.0, 70050.0, 68550.0, 69000.0, 71100.0, 74100.0, 72075.0, 75225.0, 77625.0, 75900.0, 78300.0, 78525.0, 78450.0, 75675.0, 76650.0, 78225.0, 75150.0, 74850.0, 72750.0, 71700.0, 70500.0, 75075.0, 73275.0, 73125.0, 70950.0, 70050.0, 71100.0, 69900.0, 72825.0, 74475.0, 75375.0, 75075.0, 75450.0, 75975.0, 78375.0, 78375.0, 79500.0, 78075.0, 75675.0, 74475.0, 74700.0, 76200.0, 72600.0, 70575.0, 69675.0, 69375.0, 70200.0, 72825.0, 72825.0, 71025.0, 71925.0, 79725.0, 79500.0, 81525.0, 76425.0, 75600.0, 79425.0, 81450.0, 79425.0, 79500.0, 77250.0, 74850.0, 77100.0, 82125.0, 85725.0, 83025.0, 80325.0, 79425.0, 73800.0, 72975.0, 75150.0, 72225.0, 72750.0, 76125.0, 74925.0, 78450.0, 77700.0, 77625.0, 77400.0, 78300.0, 77700.0, 76350.0, 76050.0, 74850.0, 71850.0, 76575.0, 70350.0, 69150.0, 69300.0, 68625.0, 68925.0, 69525.0, 68250.0, 70875.0, 68100.0, 68850.0, 69450.0, 66450.0, 70425.0, 68625.0, 69825.0, 71925.0, 67200.0, 70425.0, 69825.0, 70500.0, 74250.0, 68475.0, 70425.0, 68100.0, 67800.0, 65100.0, 66900.0, 70800.0, 73725.0, 73500.0, 67800.0, 69750.0, 70050.0, 69150.0, 68550.0, 68025.0, 68175.0, 64650.0, 64575.0, 68250.0, 65700.0, 67050.0, 63450.0, 60900.0, 61800.0, 64425.0, 60975.0, 61275.0, 62100.0, 60000.0, 60975.0, 63075.0, 63075.0, 68400.0, 59925.0, 64650.0, 62100.0, 63975.0, 67950.0, 64725.0, 63975.0, 61125.0, 66000.0, 66900.0, 68700.0, 66900.0, 64650.0, 62325.0, 64350.0, 64800.0, 62475.0, 60900.0, 64275.0, 62025.0, 61500.0, 62325.0, 62475.0, 61575.0, 59700.0, 60375.0, 62250.0, 56100.0, 61725.0, 57825.0, 57525.0, 59100.0, 55200.0, 54600.0, 53550.0, 53475.0, 61200.0, 56100.0, 57075.0, 57675.0, 59550.0, 63150.0]


u,m,l=talib.BBANDS(np.array(a),timeperiod=60,nbdevup=2,nbdevdn=2,matype=MA_Type.TEMA)
u1,m1,l1=talib.BBANDS(np.array(a),timeperiod=60,nbdevup=2,nbdevdn=2,matype=MA_Type.T3)
u2,m2,l2=talib.BBANDS(np.array(a),timeperiod=60,nbdevup=2,nbdevdn=2,matype=MA_Type.SMA)

u[-1]-l[-1]
u1[-1]-l1[-1]
u2[-1]-l2[-1]
```

The result I got was "14676.235041726468" in all 3 cases.
I have a feeling that your calculations are based on the classic version of bollinger bands.
Where, no matter what kind of MA is used for the middle band, the width between the bands is calculated based on the SMA values.

Is there an option you have to calculate the bands based on the MA type ??

**Comment by paritoshjchandran** [comment]
I have spent the past 3 hours looking at your C++ version of the code and you seem to have implemented it there.
There is this neat and tidy option "optInMethod" where you can set it to the classical or non classical version of the calculation.
Is there such a parameter for python function ??

**Comment by mrjbq7** [comment]
Where do you see ``optInMethod``?

This is what I see available:

```C
/*
 * TA_BBANDS - Bollinger Bands
 * 
 * Input  = double
 * Output = double, double, double
 * 
 * Optional Parameters
 * -------------------
 * optInTimePeriod:(From 2 to 100000)
 *    Number of period
 * 
 * optInNbDevUp:(From TA_REAL_MIN to TA_REAL_MAX)
 *    Deviation multiplier for upper band
 * 
 * optInNbDevDn:(From TA_REAL_MIN to TA_REAL_MAX)
 *    Deviation multiplier for lower band
 * 
 * optInMAType:
 *    Type of Moving Average
 * 
 * 
 */
TA_RetCode TA_BBANDS( int    startIdx,
                      int    endIdx,
                      const double inReal[],
                      int           optInTimePeriod, /* From 2 to 100000 */
                      double        optInNbDevUp, /* From TA_REAL_MIN to TA_REAL_MAX */
                      double        optInNbDevDn, /* From TA_REAL_MIN to TA_REAL_MAX */
                      TA_MAType     optInMAType,
                      int          *outBegIdx,
                      int          *outNBElement,
                      double        outRealUpperBand[],
                      double        outRealMiddleBand[],
                      double        outRealLowerBand[] );
```

**Comment by mrjbq7** [comment]
That looks like the unreleased "new version" of ta-lib.  This is built around wrapping the released version 0.4.0.  And it's not *my* ta-lib.  I just wrote this Python wrapper to it.

Feedback is welcome, though.  And if you have a desire for this to work with the unreleased version, then we can make a branch that's compatible pretty easily.

**Comment by paritoshjchandran** [comment]
Well, since I have sorted out this problem for myself, I have no particular need for it.

That being said, I am not sure if the next person would need this in the future or not.
I guess a branch of this would be worthwhile if similar issues are raised.

Once again, thank you for your prompt and very helpful response.
Keep up the great work, guys.
